pkg/cloud/services/networking/trunk.go

```diff
@@ -88,6 +89,29 @@ func (s *Service) DeleteTrunk(eventObject runtime.Object, portID string) error {
 	if len(trunkInfo) != 1 {
 		return nil
 	}
+	// Delete sub-ports if trunk is associated with sub-ports
+	err = wait.PollUntilContextTimeout(context.TODO(), retryIntervalSubPortDelete, timeoutTrunkDelete, true, func(_ context.Context) (bool, error) {
+		if err := s.client.RemoveTrunkSubPorts(trunkInfo[0].ID); err != nil {
+			if capoerrors.IsNotFound(err) {
+				record.Eventf(eventObject, "SuccessfulRemoveTrunkSubPorts", "Deleted sub ports Trunk %s with id %s", trunkInfo[0].Name, trunkInfo[0].ID)
+				return true, nil
+			}
+			if capoerrors.IsConflict(err) {
+				return false, nil
+			}
+			if capoerrors.IsRetryable(err) {
+				return false, nil
+			}
+			return false, err
+		}
+		return true, nil
+	})
+	if err != nil {
+		record.Warnf(eventObject, "FailedRemoveTrunkSubPorts", "Failed to delete sub ports trunk %s with id %s: %v", trunkInfo[0].Name, trunkInfo[0].ID, err)
+		return err
+	}
+
+	record.Eventf(eventObject, "SuccessfulRemoveTrunkSubPorts", "Removed trunk sub ports %s with id %s", trunkInfo[0].Name, trunkInfo[0].ID)
```

Comment:
I note that we'll always generate this event whether we removed any subports or not, but... why have a trunk if you're not going to add subports? I guess this is fine.